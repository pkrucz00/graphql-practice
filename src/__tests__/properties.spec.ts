import { describe } from "node:test";
import { expect, test } from "vitest";
import { executor } from "./setup";
import { parse } from "graphql";

describe("properties", () => {
  test("getProperties", async () => {
    const query = parse(`
      query {
        properties {
          id
          street
          city
          zip
          state
          weatherData {
            temperature
            windSpeed
          }
          coordinates {
            latitude
            longitude
          }
          createdAt
      }
    }
    `);
    const result = await executor({ document: query });
    expect(result).toMatchSnapshot();
  });
});
