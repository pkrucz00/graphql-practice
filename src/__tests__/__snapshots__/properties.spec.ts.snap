// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getProperties 1`] = `
{
  "data": {
    "properties": [
      {
        "city": "San Francisco",
        "coordinates": {
          "latitude": 37.7749,
          "longitude": -122.4194,
        },
        "createdAt": "2025-02-22",
        "id": "1",
        "state": "CA",
        "street": "123 Main St",
        "weatherData": {
          "temperature": 70,
          "windSpeed": 5,
        },
        "zip": "94105",
      },
      {
        "city": "Los Angeles",
        "coordinates": {
          "latitude": 34.0522,
          "longitude": -118.2437,
        },
        "createdAt": "2025-02-22",
        "id": "2",
        "state": "CA",
        "street": "456 Elm St",
        "weatherData": {
          "temperature": 75,
          "windSpeed": 3,
        },
        "zip": "90001",
      },
      {
        "city": "New York",
        "coordinates": {
          "latitude": 40.7128,
          "longitude": -74.006,
        },
        "createdAt": "2025-02-22",
        "id": "3",
        "state": "NY",
        "street": "789 Oak St",
        "weatherData": {
          "temperature": 65,
          "windSpeed": 10,
        },
        "zip": "10001",
      },
    ],
  },
}
`;
